File: simpleclaw_mobile/lib/models/usage.dart
class UsageData {
  const UsageData({
    required this.used,
    required this.limit,
    required this.remaining,
  });

  final double used;
  final double limit;
  final double remaining;

  factory UsageData.fromJson(Map<String, dynamic> json) {
    return UsageData(
      used: (json['used'] as num).toDouble(),
      limit: (json['limit'] as num).toDouble(),
      remaining: (json['remaining'] as num).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'used': used,
      'limit': limit,
      'remaining': remaining,
    };
  }
}

File: src/types/usage.ts
export interface UsageData {
  used: number;
  limit: number;
  remaining: number;
}

export function usageFromJson(json: Record<string, unknown>): UsageData {
  return {
    used: Number(json['used']),
    limit: Number(json['limit']),
    remaining: Number(json['remaining']),
  };
}
